addons/udes_stock/models/stock_move.py

```diff
@@ -118,6 +131,42 @@ def _unreserve_initial_demand(self, new_move):
             {"move_id": new_move.id, "product_uom_qty": 0}
         )
 
+    def propagate_cancellation_to_next_picks(self, old_dest_ids):
+        """
+        Propagate cancelled qtys to move_dest_ids.
+        If a full qty match is found, cancel that move - calling this function for that move
+        If a part qty match is found, decrement the qty from the move, and continue down the chain
+
+        `propagate_cancel` on `stock.rule` could be used, however it is not visible on `Push` rules
+        and does not appear to propagate partial cancellations.
+
+        :param: old_dest_ids: stock.move(x) recordset(s), fallback if no move_dest_ids in self.
+                              Can be emptyset
+        """
+        StockMove = self.env["stock.move"].browse()
+        next_moves_filter = lambda m: m.picking_type_id.u_propagate_cancel == True
+        next_moves = self.move_dest_ids.filtered(next_moves_filter)
+        if not next_moves:
+            # NB: This handles a symptom of Issue #2771
+            next_moves = old_dest_ids
+        qty = self.product_uom_qty
+        while next_moves:
+            fully_matched_move = next_moves.filtered(lambda m: m.product_uom_qty == qty)
+            for move in fully_matched_move:
+                move._action_cancel()
+                # Recurse via action_cancel, we don't need to continue through the chain here.
+                next_moves = StockMove
+            else:
+                next_moves_todo = next_moves
+                next_moves = StockMove  # Clear this as we potentially rebuild inside the loop
+                for move in next_moves_todo:  # Iterate on the non-cleared set
+                    if move.product_uom_qty > qty:
+                        move.product_uom_qty -= qty
+                        # Recursing inside here is only necessary for part cancellations
+                        next_moves |= move.move_dest_ids.filtered(next_moves_filter)
+                    else:  # qty is less, as there was no fully_matched_move
```

Comment:
This else statement is for when `move.product_uom_qty < qty` right? so I think the comment should read: "qty is greater"

Reply:
Yeah i think being more explicit is good
```python
else:  # move qty is less, as there was no fully_matched_move
```